### ISSUE TITLE
trash: error -1701

### ISSUE THREAD

**Issue by ghost** [ali-rantakari/trash#37]
I am getting this error when calling a command in a script. The same command called in a shell doesn't receive the error.

Shell script being called by another shell script with "$1" being a single quoted path.
```
#!/bin/bash
#find and delete non-english english subtitles
{
  /usr/bin/find "$1" -type f -iname "*.srt" | grep -v -e "\\.eng" -e "].srt" -e "]\.[0-9].srt" | while read -r SUBS; do trash -F "$SUBS"; done
} >> /Users/john/Desktop/QB-FB_test/findAndDelete_subs.txt 2>&1

```
Result:
```
+ trash -F '/Volumes/PlexMedia/PlexServer_1/Movies/Test Folder (1968)/moviename.swe.srt'
trash: error -1701
+ read -r SUBS
```

Command line
```
$ set -- "/Volumes/PlexMedia/PlexServer_1/Movies/Test Folder (1968)"
$ { 
 /usr/bin/find "$1" -type f -iname "*.srt" | grep -v -e "\\.eng" -e "].srt" -e "]\.[0-9].srt" | while read -r SUBS; do trash -F "$SUBS"; done \
 } >> /Users/john/Desktop/QB-FB_test/findAndDelete_subs.txt 2>&1
```

This results in the files being sent to the trash

**Comment by deus0ww** [comment]
Try adding the calling script and/or the 'trash' binary to: System Preferences -> Security & Privacy -> Privacy -> Accessibility .

**Comment by ghost** [comment]
After 15 days I decided to just install another `trash` binary and move on. Maybe someone else would like to test this and see if that is indeed what causes error -1701. It may be.  It would be good if this binary was updated to mention the modern macOS security hurdles.

**Comment by Mellbourn** [comment]
@jprokos agreed, I started using https://github.com/morgant/tools-osx/blob/master/src/trash, it's much better: 
Delete using `Finder` is the default (so `Put back` still works). If `Finder` fails, it quietly falls backs to non-finder delete so it works anyway.
IMHO the `ali-rantakari/trash` should not be the one installed by `brew install trash`

**Comment by gingerbeardman** [comment]
Came here with this problem.

**Comment by gsbabil** [comment]
- FWIW, I noticed the `-1701` error appears only in `Kitty.app` for me, but not in MacOS's built-in `Terminal.app`.

- Another observation is that when I have `trash` aliased as `rm` _e.g._ with `alias rm="trash -v -F"` in my `~/.zshrc`, I get `-1701` error when I do `rm file-path`, but not when I do `trash file-path`.

    - I have both `/usr/local/bin/{trash,zsh}`, `Terminal.app` and `Kitty.app` added in System Preferences -> Security & Privacy -> Privacy -> Accessibility.

Thank you.

#### ## EDIT [04-May-2022]

- ✅ Looks like I was able to fix the problem after running `tccutil reset AppleEvents` 
  - From `Kitty.app` I ran the command above and then invoked `trash <file>`, which presents the `"Apple Events"` permission prompt, and everything seems to be working after approving the permission. 
  
Hope this helps!

**Comment by veloxo** [comment]
I solved this for `trash -F` by re-enabling _Finder_ permissions under _System Settings → Privacy & Security → Automation → iTerm_ (which I had manually disabled at some point).

**Comment by ScArLeXiA** [comment]
I solved the problem with the help of @gsbabil and @veloxo comments. I share my solution steps.

In Sonoma 14.5, `System Settings > Privacy & Security > Automation` must still contain the Finder control access permission for terminal app to execute `trash -F`.

I use [Tabby](https://tabby.sh) as a terminal app, but when I execute `trash -F`, the permission request dialog does not appear.
It appears that Automation permission must be requested from the app and cannot be added from System Settings.

The following is how to request it manually.

```bash
# Reset Automation permission for the bundle ID of the app currently at the frontmost (i.e., terminal app) to ensure the appearance of the permission request dialog.
tccutil reset AppleEvents "$(osascript -e 'id of app (path to frontmost application as text)')"

# Use the Finder application to get the name of the item in the trash (this opens a permission request dialog).
osascript -e 'tell application "Finder" to get name of items in trash'
```

![image](https://github.com/user-attachments/assets/a1bc8dac-b486-4134-9f21-745d8e48376c)
![image](https://github.com/user-attachments/assets/1d00cfe0-1e44-4715-b37d-f5f5ef60dfc6)

Finally, by allowing the request, I was able to successfully execute `trash -F`.
